Ruff does not enforce the number of blank lines between functions, classes, and methods

### Description

This is a regression from #8265.

### Steps to reproduce

```console
$ cat foo.py 
def foo():
    pass
def bar():
    pass
$ flake8 foo.py 
foo.py:3:1: E302 expected 2 blank lines, found 1
$ ruff check foo.py 
All checks passed!
```

In Ruff such rules are under preview:

```console
$ ruff check --preview foo.py 
E302 [*] Expected 2 blank lines, found 0
 --> foo.py:3:1
  |
1 | def foo():
2 |     pass
3 | def bar():
  | ^^^
4 |     pass
  |
help: Add missing blank line(s)
1 | def foo():
2 |     pass
3 + 
4 + 
5 | def bar():
6 |     pass

Found 1 error.
[*] 1 fixable with the `--fix` option.
```

### Rucio Version

_No response_

### Additional Information

_No response_

--- a/bin/rucio
+++ b/bin/rucio
@@ -102,7 +102,7 @@ def map_legacy_command() -> Optional[list[str]]:
 def run_command(args, use_multi_host_functionality):
     commands = ("account", "config", "did", "replica", "rse", "rule", "scope", "subscription", "ping", "whoami", "test-server", "lifetime-exception", "upload", "download", "opendata")
     if (any(arg in commands for arg in sys.argv)) or args.help or args.version:
-        main(standalone_mode = not use_multi_host_functionality)  # pylint: disable=E1120
+        main(standalone_mode=not use_multi_host_functionality)  # pylint: disable=E1120
 
     else:
         try:
@@ -119,7 +119,7 @@ def run_command(args, use_multi_host_functionality):
             else:
                 sys.argv = ["rucio", "-h"]
 
-                main(standalone_mode = not use_multi_host_functionality)  # pylint: disable=E1120
+                main(standalone_mode=not use_multi_host_functionality)  # pylint: disable=E1120
 
 
 if __name__ == "__main__":

--- a/bin/rucio
+++ b/bin/rucio
@@ -124,7 +124,6 @@ def run_command(args, use_multi_host_functionality):
 
 if __name__ == "__main__":
 
-
     parser = argparse.ArgumentParser(add_help=False)
     # Check for legacy flag
     parser.add_argument("--legacy", action="store_true")

--- a/bin/rucio
+++ b/bin/rucio
@@ -102,7 +102,7 @@ def map_legacy_command() -> Optional[list[str]]:
 def run_command(args, use_multi_host_functionality):
     commands = ("account", "config", "did", "replica", "rse", "rule", "scope", "subscription", "ping", "whoami", "test-server", "lifetime-exception", "upload", "download", "opendata")
     if (any(arg in commands for arg in sys.argv)) or args.help or args.version:
-            main(standalone_mode = not use_multi_host_functionality)  # pylint: disable=E1120
+        main(standalone_mode = not use_multi_host_functionality)  # pylint: disable=E1120
 
     else:
         try:

--- a/bin/rucio
+++ b/bin/rucio
@@ -98,6 +98,7 @@ def map_legacy_command() -> Optional[list[str]]:
 
     return new_command
 
+
 def run_command(args, use_multi_host_functionality):
     commands = ("account", "config", "did", "replica", "rse", "rule", "scope", "subscription", "ping", "whoami", "test-server", "lifetime-exception", "upload", "download", "opendata")
     if (any(arg in commands for arg in sys.argv)) or args.help or args.version:

--- a/lib/rucio/cli/account.py
+++ b/lib/rucio/cli/account.py
@@ -301,7 +301,6 @@ def identity_add(ctx: click.Context, account_name: str, type_: str, id: str, ema
     print('Added new identity to account: %s-%s' % (id, account_name))
 
 
-
 @identity.command("remove")
 @click.argument("account-name", required=True)
 @click.option("--type", "type_", type=click.Choice(["X509", "GSS", "USERPASS", "SSH", "SAML", "OIDC"]), help="Authentication type", required=True)

--- a/lib/rucio/cli/account.py
+++ b/lib/rucio/cli/account.py
@@ -259,6 +259,7 @@ def limit_remove(ctx: click.Context, account_name: str, rse: str, locality: str)
     ctx.obj.client.delete_account_limit(account=account_name, rse=rse, locality=locality)
     print('Deleted account limit for account %s and RSE %s' % (account_name, rse))
 
+
 @account.group("identity")
 def identity():
     """Manage identities for an account - used to login"""

--- a/lib/rucio/cli/utils.py
+++ b/lib/rucio/cli/utils.py
@@ -17,7 +17,7 @@ import json
 import logging
 import os
 import signal
-import subprocess
+import subprocess  # noqa: S404 -- subprocess used for external commands
 import sys
 import traceback
 from configparser import NoOptionError, NoSectionError

--- a/lib/rucio/client/downloadclient.py
+++ b/lib/rucio/client/downloadclient.py
@@ -21,7 +21,7 @@ import random
 import secrets
 import shutil
 import signal
-import subprocess
+import subprocess  # noqa: S404 -- subprocess used for external commands
 import time
 from queue import Empty, Queue, deque
 from threading import Thread

--- a/lib/rucio/common/dumper/consistency.py
+++ b/lib/rucio/common/dumper/consistency.py
@@ -16,7 +16,7 @@ import datetime
 import logging
 import os
 import re
-import subprocess
+import subprocess  # noqa: S404 -- subprocess used for external commands
 import tempfile
 from typing import TYPE_CHECKING, Any, Optional, Union, cast
 

--- a/lib/rucio/common/pcache.py
+++ b/lib/rucio/common/pcache.py
@@ -19,7 +19,7 @@ import getopt
 import os
 import re
 import signal
-import subprocess
+import subprocess  # noqa: S404 -- subprocess used for external commands
 import sys
 import time
 from socket import gethostname

--- a/lib/rucio/common/utils.py
+++ b/lib/rucio/common/utils.py
@@ -27,7 +27,7 @@ import os.path
 import re
 import signal
 import socket
-import subprocess
+import subprocess  # noqa: S404 -- subprocess used for external commands
 import tempfile
 import threading
 import time

--- a/lib/rucio/common/utils.py
+++ b/lib/rucio/common/utils.py
@@ -40,7 +40,7 @@ from itertools import zip_longest
 from typing import TYPE_CHECKING, Any, Optional, TypeVar, Union, cast
 from urllib.parse import parse_qsl, quote, urlencode, urlparse, urlunparse
 from uuid import uuid4 as uuid
-from xml.etree import ElementTree
+from xml.etree import ElementTree  # noqa: S405 -- trusted XML input
 
 import requests
 from typing_extensions import ParamSpec

--- a/lib/rucio/core/oidc.py
+++ b/lib/rucio/core/oidc.py
@@ -15,7 +15,7 @@
 import hashlib
 import json
 import logging
-import subprocess
+import subprocess  # noqa: S404 -- subprocess used for external commands
 import traceback
 from datetime import datetime, timedelta
 from math import floor

--- a/lib/rucio/daemons/auditor/hdfs.py
+++ b/lib/rucio/daemons/auditor/hdfs.py
@@ -17,7 +17,7 @@ import logging
 import os
 import re
 import shutil
-import subprocess
+import subprocess  # noqa: S404 -- subprocess used for external commands
 import tempfile
 from typing import TYPE_CHECKING
 

--- a/lib/rucio/rse/protocols/gfal.py
+++ b/lib/rucio/rse/protocols/gfal.py
@@ -17,7 +17,7 @@ import json
 import logging
 import os
 import re
-import subprocess
+import subprocess  # noqa: S404 -- subprocess used for external commands
 import urllib.parse as urlparse
 from threading import Timer
 

--- a/lib/rucio/rse/protocols/posix.py
+++ b/lib/rucio/rse/protocols/posix.py
@@ -16,7 +16,7 @@ import logging
 import os
 import os.path
 import shutil
-from subprocess import call
+from subprocess import call  # noqa: S404 -- subprocess used for external commands
 
 from rucio.common import exception
 from rucio.common.checksum import adler32

--- a/lib/rucio/rse/protocols/srm.py
+++ b/lib/rucio/rse/protocols/srm.py
@@ -15,7 +15,7 @@
 import os
 import re
 import urllib.parse as urlparse
-from subprocess import getstatusoutput
+from subprocess import getstatusoutput  # noqa: S404 -- subprocess used for external commands
 
 from rucio.common import exception
 from rucio.common.utils import execute

--- a/lib/rucio/rse/protocols/storm.py
+++ b/lib/rucio/rse/protocols/storm.py
@@ -14,7 +14,7 @@
 
 import logging
 import os
-from xml.dom import minidom
+from xml.dom import minidom  # noqa: S408 -- trusted XML input
 
 import requests
 

--- a/lib/rucio/rse/protocols/webdav.py
+++ b/lib/rucio/rse/protocols/webdav.py
@@ -18,7 +18,7 @@ import sys
 from dataclasses import dataclass
 from typing import Any, Optional
 from urllib.parse import urlparse
-from xml.etree import ElementTree
+from xml.etree import ElementTree  # noqa: S405 -- trusted XML input
 
 import requests
 from requests.adapters import HTTPAdapter

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -157,6 +157,7 @@ exclude = [
 ]
 
 [tool.ruff.lint]
+preview = true  # Go to https://docs.astral.sh/ruff/rules/ to see which rules are additionally added
 select = [
     "E",    # All default pycodestyle error families
     "W",    # All default pycodestyle warning families

--- a/setuputil.py
+++ b/setuputil.py
@@ -11,7 +11,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import subprocess
+import subprocess  # noqa: S404 -- subprocess used for external commands
 import sys
 from typing import Union
 

--- a/tests/test_replica.py
+++ b/tests/test_replica.py
@@ -19,7 +19,7 @@ from datetime import datetime, timedelta
 from json import dumps
 from typing import TYPE_CHECKING
 from unittest import mock
-from xml.etree import ElementTree
+from xml.etree import ElementTree  # noqa: S405 -- trusted XML input
 
 import pytest
 import xmltodict

--- a/tools/generate-release-notes.py
+++ b/tools/generate-release-notes.py
@@ -14,7 +14,7 @@
 # limitations under the License.
 
 import json
-import subprocess
+import subprocess  # noqa: S404 -- subprocess used for external commands
 import sys
 
 import requests

--- a/tools/generate_version.py
+++ b/tools/generate_version.py
@@ -20,7 +20,7 @@ base_path = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 sys.path.append(base_path)
 os.chdir(base_path)
 
-import subprocess  # noqa: E402
+import subprocess  # noqa: E402,S404 -- subprocess used for external commands
 
 
 def run_git_command(cmd):

--- a/tools/run_pyright/generate.py
+++ b/tools/run_pyright/generate.py
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 import json
-import subprocess
+import subprocess  # noqa: S404 -- subprocess used for external commands
 import sys
 from pathlib import Path
 from typing import TYPE_CHECKING, Any

--- a/tools/test/run_tests.py
+++ b/tools/test/run_tests.py
@@ -85,7 +85,7 @@ import multiprocessing
 import os
 import pathlib
 import shutil
-import subprocess
+import subprocess  # noqa: S404 -- subprocess used for external commands
 import sys
 import time
 import traceback

